### DISCUSSION TITLE
Week 5

### DISCUSSION THREAD

**Discussion by Wildeswiesel** [Wildeswiesel/Space-Waffel#7] [Announcements]
Zuerst haben wir am anfang der Stunde einen kleinen Test gemacht [diesen finden sie hier](https://docs.google.com/forms/d/e/1FAIpQLSca_WGQ-PbjKKy92bYFjyY0dBUUzWwMwqPHyA660ah_D4jJTw/viewform)

Dann haben wir uns bei [w3schools](https://www.w3schools.com)

![image](https://github.com/Wildeswiesel/Space-Waffel/assets/152853691/9636e123-7771-4950-aa6d-b93ff7650de6)

Danach wurden wir beauftragt das min und max und dupikate auszuschließen
```
mylist = [1,3,2,4,6,8,11,99,6,1]
print(max(mylist))

mylist = list( dict.fromkeys(mylist) )
print(mylist)
```
den link finden sie [hier](https://github.com/Wildeswiesel/Space-Waffel/tree/Test/DHBW/Python%20Unterricht%20am%2015.01). Dort finden sie auch alle andere

Danach haben wir eine Transaction Liste erstellt welche uns verschiedene Sachen ausgegebn hat
Liste:
```
transactions = [
    {'data': 'purchase', 'amount': 50, 'date':'2024-01-14'},
    {'data': 'sale', 'amount': 540, 'date':'2024-01-15'},
    {'data': 'purchase', 'amount': 250, 'date':'2024-01-14'},
]
```
Rausgekommen ist

Einnahmen = 300
Ausgaben = 540
Sie haben geld verloren
Wie viele Transaktionen wurden am 14.01 durchgeführt: 2

Danach haben wir mittels Regex noch das Datums Vormat festgelegt 
```

import re

def is_valid_date_format(date_string):
    date_pattern = re.compile(r"^\d{4}-\d{2}-\d{2}$")
    return bool(date_pattern.match(date_string))


my_date = my_transactions[0]["date"]
print(my_transactions[0]["date"])
print(type(my_date))
print(is_valid_date_format(my_date))
```
<hr>

Daheim habe ich noch (weil mich das CapsLock immer nervt) mittels AHK die CapsLock taste gesperrt und es kommt statt dessen ein "/" 
Der den Link finden sie [hier](https://github.com/Wildeswiesel/Space-Waffel/blob/Test/Projekte/AHK/CapsLocked.ahk)
Und da ich mal meine Festplatte aufräumen wollte habe ich mir ein Code geschrieben der alle datein bis auf bestimmte löscht 
Link finden Sie [hier](https://github.com/Wildeswiesel/Space-Waffel/blob/Test/Projekte/Python/delet%20data.py)

Und ein paar runden codinggames am Tag waren auch noch drin

**Comment by stechmo** [comment]
Schöner Einblick auf deine Coding-Woche!

**Comment by SvenSrc** [comment]
Du könntest auch deine CapsLock Taste entfernen. Aber sonst gute Idee

**Reply by Wildeswiesel** [reply]
ja stimmt allerdings hab ich mir gedacht man braucht ab und zu die / taste um bei manchen Programmiersprachen etwas auszuklammern

**Comment by HamidrezaRahimian** [comment]
really nice and entertaining guide ,haven't even noticed to time while reading it , great job! 
will check next week as well , keep it up!
